pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -117,6 +137,34 @@ type ClusterWorkspaceTypeSpec struct {
 	//
 	// +optional
 	AdditionalWorkspaceLabels map[string]string `json:"additionalWorkspaceLabels,omitempty"`
+
+	// defaultChildWorkspaceType is the ClusterWorkspaceType that will be used
+	// by default if another, nested ClusterWorkspace is created in a workspace
+	// of this type. When this field is unset, the user must specify a type when
+	// creating nested workspaces.
+	//
+	// +optional
+	DefaultChildWorkspaceType string `json:"defaultChildWorkspaceType,omitempty"`
+
+	// allowedChildWorkspaceTypes is a list of ClusterWorkspaceTypes that can be
+	// created in a workspace of this type.
+	//
+	// By default, no type is allowed. This means no other workspace can be nested
+	// within a workspace of the given type. The name `*` allows any child type to
+	// be nested.
+	//
+	// +optional
+	AllowedChildWorkspaceTypes []string `json:"allowedChildWorkspaceTypes,omitempty"`
+
+	// allowedParentWorkspaceTypes is a list of ClusterWorkspaceTypes that this type
+	// can be created in.
+	//
+	// By default, no type is allowed. This means no other workspace can have a
+	// workspace of the given type nested inside it. The name `*` allows any parent
+	// type to nest this one.
+	//
+	// +optional
+	AllowedParentWorkspaceTypes []string `json:"allowedParentWorkspaceTypes,omitempty"`
```

Comment:
`minItems: 1` zero does not make sense, does it?

Reply:
Where did you see that? What?

Reply:
`AllowedParentWorkspaceTypes == empty` means it cannot be used. Hence, we should have at least one allowed parent type.